### ISSUE TITLE
IDynamicInterfaceCastable and dynamic keyword

### ISSUE THREAD

**Issue by jkotas** [dotnet/runtime#37812]
dynamic keyword does not work with IDynamicInterfaceCastable

cc @elinor-fung

**Comment by AaronRobinsonMSFT** [comment]
Given our focused scenarios for the `IDynamicInterfaceCastable` interface I am inclined to say we should defer this to the future. @elinor-fung if enabling this is relatively simple - 1 or 2 days - then lets do it. Otherwise I am in favor of moving this to .NET 6.

**Comment by elinor-fung** [comment]
I do not think this would be a 1-2 day affair. We don't know all the interfaces supported by the `IDynamicInterfaceCastable` or what interface the user is trying to use. The support of `dynamic` with COM objects (which faces a similar problem) relies on `IDispatch` being implemented - we would need some mechanism like that for support with `IDynamicInterfaceCastable`.

Unless we have a compelling use-case now, I would defer this until we have demand/requests for it.
